src/optimizer.cpp

```diff
@@ -101,19 +103,11 @@ geometry_msgs::msg::TwistStamped Optimizer::evalControl(
   const geometry_msgs::msg::Twist & robot_speed,
   const nav_msgs::msg::Path & plan, nav2_core::GoalChecker * goal_checker)
 {
-  state_.pose = robot_pose;
-  state_.speed = robot_speed;
-
-  auto plan_tensor = utils::toTensor(plan);
+  prepare(robot_pose, robot_speed, plan, goal_checker);
 
-  costs_.fill(0);
-  bool stop_flag = false;
   for (size_t i = 0; i < settings_.iteration_count; ++i) {
     generated_trajectories_ = generateNoisedTrajectories();
-    auto data =
-      models::CriticFunctionData{state_, generated_trajectories_,
```

Comment:
`Don't you need to update the trajectories, since each iteration will have new `generated_trajectories_`? Preparing once before the loop wouldn't update for multiple iterations.

Reply:
Notice, that i don't actually update generated_trajectories_ before the loop. Now data is a member variable that keeps a reference to generated_trajectories_

Reply:
Ah, good point